lib/Doctrine/MongoDB/Aggregation/Stage/Match.php

```diff
@@ -0,0 +1,604 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace Doctrine\MongoDB\Aggregation\Stage;
+
+use Doctrine\MongoDB\Aggregation\Builder;
+use Doctrine\MongoDB\Aggregation\Stage;
+use Doctrine\MongoDB\Query\Expr;
+use GeoJson\Geometry\Geometry;
+
+/**
+ * Fluent interface for adding a $match stage to an aggregation pipeline.
+ *
+ * @author alcaeus <alcaeus@alcaeus.org>
+ */
+class Match extends Stage
+{
+    /**
+     * @var Expr
+     */
+    protected $query;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __construct(Builder $builder)
+    {
+        parent::__construct($builder);
+
+        $this->query = new Expr();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getExpression()
+    {
+        return array(
+            '$match' => $this->query->getQuery()
+        );
+    }
+
+    /**
+     * Add an $and clause to the current query.
+     *
+     * You can create a new expression using the {@link Builder::expr()} method.
+     *
+     * @see Expr::addAnd()
+     * @see http://docs.mongodb.org/manual/reference/operator/and/
+     * @param array|Expr $expression
+     * @return self
+     */
+    public function addAnd($expression)
```

Comment:
At a glance, it looks like this class is proxying all of the methods on the query builder. Would it make more sense to define a simple `__call()` magic method that allows a whitelist of method names to be proxied (passing all received arguments along in the process)? I suggest a whitelist because there are probably certain methods on the query builder that we don't want exposed (such as execution).